py4j-java/src/main/java/py4j/CallbackClient.java

```diff
@@ -176,9 +178,9 @@ public CallbackClient(int port, InetAddress address, long minConnectionTime, Tim
 	 *            Python program is closed.
 	 */
 	public CallbackClient(int port, InetAddress address, long minConnectionTime, TimeUnit minConnectionTimeUnit,
-			SocketFactory socketFactory, boolean enableMemoryManagement, int readTimeout) {
+			SocketFactory socketFactory, boolean enableMemoryManagement, int readTimeout, int createSocketTimeout) {
```

Comment:
To avoid an API break, probably want to keep a constructor of the existing signature and use default socket connect timeout.

```
public CallbackClient(int port, InetAddress address, long minConnectionTime, TimeUnit minConnectionTimeUnit,
			SocketFactory socketFactory, boolean enableMemoryManagement, int readTimeout) {
    this(port, address, null, minConnectionTime, minConnectionTimeUnit, socketFactory, enableMemoryManagement,
				readTimeout, GatewayServer.DEFAULT_CONNECT_TIMEOUT);
}
```